geomstats/geometry/discrete_curves.py

```diff
@@ -440,3 +440,150 @@ def dist(self, point_a, point_b):
+        atol : float
+            Tolerance of the projection algorithm.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        proj : array-like, shape=[..., n_sampling_points, ambient_dim]
+        """
+        is_euclidean = isinstance(self.ambient_manifold, Euclidean)
+        is_planar = is_euclidean and self.ambient_manifold.dim == 2
+
+        if not is_planar:
+            raise AssertionError('The projection is only implemented '
+                                 'for discrete curves embedded in a'
+                                 '2D Euclidean space.')
+
+        srv_metric = self.square_root_velocity_metric
+        srv = srv_metric.square_root_velocity(curve)[0]
+        srv_proj = srv_metric.project_srv(srv, atol=atol)
+        proj = srv_metric.square_root_velocity_inverse(srv_proj,
+                                                       gs.array([curve[0]]))
+
+        return proj
+
+
+class ClosedSRVMetric(SRVMetric):
+    """Elastic metric on closed curves.
+
+    See [Sea2011]_ for details.
+
+    Parameters
+    ----------
+    ambient_manifold : Manifold
+        Manifold in which curves take values.
+
+    References
+    ----------
+    .. [Sea2011] A. Srivastava, E. Klassen, S. H. Joshi and I. H. Jermyn,
+    "Shape Analysis of Elastic Curves in Euclidean Spaces,"
+    in IEEE Transactions on Pattern Analysis and Machine Intelligence,
+    vol. 33, no. 7, pp. 1415-1428, July 2011.
+    """
+
+    def __init__(self, ambient_manifold):
+        super(ClosedSRVMetric, self).__init__(ambient_manifold)
+
+    def project_srv(self, srv, atol=gs.atol):
+        """Project a point in the srv space into the space of closed curves srv.
+
+        Parameters
+        ----------
+        srv : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+        atol : float
+            Tolerance of the projection algorithm.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        proj : array-like, shape=[..., n_sampling_points, ambient_dim]
+        """
+        is_euclidean = isinstance(self.ambient_metric, EuclideanMetric)
+        is_planar = is_euclidean and self.ambient_metric.dim == 2
+
+        if not is_planar:
+            raise AssertionError('The projection is only implemented '
+                                 'for discrete curves embedded in a'
+                                 '2D Euclidean space.')
+
+        n_sampling_points = srv.shape[-2]
+        srv_metric = self.l2_metric(n_sampling_points)
+        ambient_inner_prod = self.ambient_metric.inner_product
+
+        def G(srv, srv_norms):
+            return gs.sum(srv * srv_norms[:, None], axis=0)
+
+        initial_norm = srv_metric.norm(srv)
+        proj = srv
+        proj_norms = self.ambient_metric.norm(proj)
+        criteria = atol + 1
+        residual = G(proj, proj_norms)
+
+        while(criteria >= atol):
+            J = gs.zeros((2, 2))
```

Comment:
2 refers to the ambient dimension right? maybe use `ambient_manifold.dim` here, so that 2 is not hard coded.